beta-src/src/state/game/game-api-slice.ts

```diff
@@ -702,9 +755,13 @@ const gameApiSlice = createSlice({
               contextKey: newContextKey,
             };
           }
+
           Object.entries(orders).forEach(([id, value]) => {
             if (value.status === "Complete") {
               state.ordersMeta[id].saved = true;
+              if (state.ordersMeta[id].update?.type === "Disband") {
+                updateUnitsDisbanding(state);
```

Comment:
You're calling the `updateUnitsDisbanding` on every single order call, updateUnitsDisbanding already loops through every order. Also, `updateUnitsDisbanding` loops through currentOrders, not ordersMeta, so what if ordersMeta changes? Will this affect or overwrite previous orders made and saved or ordersMeta?

Reply:
updateUnitsDisbanding checks currentOrders and ordersMeta for disbanded code. This is to catch updating orders and orders that are disbanded in currentOrders after the page is refreshed or revisited

Reply:
What I'm asking is, why do we need to include this loop inside this ordersMeta loop, if it already loops in the function?

Reply:
This code was added so that the ui gets updated for all disbanding units the second save button gets hit. Without this code, the unit wont show as disbanded until the page is refreshed. Remember we discussed this before and you told me to call updateUnitsDisbanding in the reducer that is listening to orders saved in gameapislice?

Reply:
I'm not saying you should remove this code, I am saying look at the placement. Do you see nothing wrong with the placement?